lib/mpl_toolkits/axes_grid1/axes_rgb.py

```diff
@@ -72,20 +73,60 @@ def imshow_rgb(ax, r, g, b, **kwargs):
     return im_rgb
 
 
-from .mpl_axes import Axes
-
 class RGBAxesBase(object):
-
+    """base class for a 4-panel imshow (RGB, R, G, B)
+
+    Layout:
+    +---------------+-----+
+    |               |  R  |
+    +               +-----+
+    |      RGB      |  G  |
+    +               +-----+
+    |               |  B  |
+    +---------------+-----+
+
+    Attributes
+    ----------
+    _defaultAxesClass : matplotlib.axes.Axes
+        defaults to 'Axes' in RGBAxes child class.
+        No default in abstract base class
+    RGB : _defaultAxesClass
+        The axes object for the three-channel imshow
+    R : _defaultAxesClass
+        The axes object for the red channel imshow
+    G : _defaultAxesClass
+        The axes object for the green channel imshow
+    B : _defaultAxesClass
+        The axes object for the blue channel imshow
+    """
     def __init__(self, *kl, **kwargs):
+        """
+        Parameters
+        ----------
+        pad : float
+            fraction of the axes height to put as padding.
+            defaults to 0.0
+        add_all : bool
+            True: Add the {rgb, r, g, b} axes to the figure
+            defaults to True.
+        axes_class : matplotlib.axes.Axes
+
+        kl :
+            Unpacked into axes_class() init for RGB
+        kwargs :
+            Unpacked into axes_class() init for RGB, R, G, B axes
+        """
         pad = kwargs.pop("pad", 0.0)
         add_all = kwargs.pop("add_all", True)
-        axes_class = kwargs.pop("axes_class", None)
-
-
-
-
-        if axes_class is None:
-            axes_class = self._defaultAxesClass
+        try:
+            axes_class = kwargs.pop("axes_class", self._defaultAxesClass)
+        except AttributeError:
+            new_msg = ("A subclass of RGBAxesBase must have a "
```

Comment:
hmmm. ```_defaultAxesClass``` *is* private, in general I'm not in favour of this defensive style, but I accept that this message would have been very helpful for you at some point. I wont hold the PR up for this though.